homeassistant/components/upnp/sensor.py

```diff
@@ -1,275 +1,257 @@
-from homeassistant.core import callback
+from homeassistant.const import DATA_BYTES, DATA_RATE_KIBIBYTES_PER_SECOND
 from homeassistant.helpers import device_registry as dr
 from homeassistant.helpers.dispatcher import async_dispatcher_connect
 from homeassistant.helpers.entity import Entity
 from homeassistant.helpers.typing import HomeAssistantType
-from homeassistant.util import Throttle
-import homeassistant.util.dt as dt_util
-
-from .const import DOMAIN as DOMAIN_UPNP, SIGNAL_REMOVE_SENSOR
-
-_LOGGER = logging.getLogger(__name__)
-
-BYTES_RECEIVED = "bytes_received"
-BYTES_SENT = "bytes_sent"
-PACKETS_RECEIVED = "packets_received"
-PACKETS_SENT = "packets_sent"
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import (
+    BYTES_RECEIVED,
+    BYTES_SENT,
+    DATA_PACKETS,
+    DATA_RATE_PACKETS_PER_SECOND,
+    DOMAIN,
+    KIBIBYTE,
+    LOGGER as _LOGGER,
+    PACKETS_RECEIVED,
+    PACKETS_SENT,
+    SIGNAL_REMOVE_DEVICE,
+    TIMESTAMP,
+)
+from .device import Device
 
 SENSOR_TYPES = {
-    BYTES_RECEIVED: {"name": "bytes received", "unit": DATA_BYTES},
-    BYTES_SENT: {"name": "bytes sent", "unit": DATA_BYTES},
-    PACKETS_RECEIVED: {"name": "packets received", "unit": "packets"},
-    PACKETS_SENT: {"name": "packets sent", "unit": "packets"},
+    BYTES_RECEIVED: {
+        "device_value_key": BYTES_RECEIVED,
+        "name": DATA_BYTES + " received",
+        "unit": DATA_BYTES,
+        "unique_id": BYTES_RECEIVED,
+        "derived_name": DATA_RATE_KIBIBYTES_PER_SECOND + " received",
+        "derived_unit": DATA_RATE_KIBIBYTES_PER_SECOND,
+        "derived_unique_id": "KiB/sec_received",
+    },
+    BYTES_SENT: {
+        "device_value_key": BYTES_SENT,
+        "name": DATA_BYTES + " sent",
+        "unit": DATA_BYTES,
+        "unique_id": BYTES_SENT,
+        "derived_name": DATA_RATE_KIBIBYTES_PER_SECOND + " sent",
+        "derived_unit": DATA_RATE_KIBIBYTES_PER_SECOND,
+        "derived_unique_id": "KiB/sec_sent",
+    },
+    PACKETS_RECEIVED: {
+        "device_value_key": PACKETS_RECEIVED,
+        "name": DATA_PACKETS + " received",
+        "unit": DATA_PACKETS,
+        "unique_id": PACKETS_RECEIVED,
+        "derived_name": DATA_RATE_PACKETS_PER_SECOND + " received",
+        "derived_unit": DATA_RATE_PACKETS_PER_SECOND,
+        "derived_unique_id": "packets/sec_received",
+    },
+    PACKETS_SENT: {
+        "device_value_key": PACKETS_SENT,
+        "name": DATA_PACKETS + " sent",
+        "unit": DATA_PACKETS,
+        "unique_id": PACKETS_SENT,
+        "derived_name": DATA_RATE_PACKETS_PER_SECOND + " sent",
+        "derived_unit": DATA_RATE_PACKETS_PER_SECOND,
+        "derived_unique_id": "packets/sec_sent",
+    },
 }
 
-IN = "received"
-OUT = "sent"
-KIBIBYTE = 1024
-
-MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=30)
+SCAN_INTERVAL = timedelta(seconds=30)
 
 
 async def async_setup_platform(
```

Comment:
We can remove this function.

Reply:
Still being used for `DataUpdateCoordinator`. Renamed to `UPDATE_INTERVAL`.